src/main/java/uk/gov/hmcts/reform/professionalapi/controller/request/DxAddressCreationRequestValidatorImpl.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.stereotype.Service;
+import org.springframework.stereotype.Component;
 import org.springframework.util.CollectionUtils;
 
-@Service
+@Component

```

Comment:
You are injecting this class into a RESTcontroller and storing state in it. That is a major no-no. Because you will have various calls to your RESTcontroller sharing the same state which will create race conditions and all sorts of unexpected behavior when there are multiple calls made. Please remove @Component and re-architect your design. The stateful members need to become part of the arguments that get passed into the validator via the RESTcontroller. If it is stateless, only then can you use the @Component syntax. So basically you have two options:
1) Remove @Component and create a local instance of the validator within the RESTcontroller method that needs it
OR
2) Keep @Component and remove all statefullness (i.e. no member variables allowed unless they are final primatives or stateless helper classes such as static utility methods)

Both options are valid. But what you have here will fail when there are multiple requests happening in succession as it is NOT thread safe.